### PR TITLE
Add multiple pattern comprehensions handling in Return clause.

### PR DESCRIPTION
**Pull request by Darych**
### Description

Add support for multiple pattern comprehensions inside Return clause.

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[Release note text]**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/query/plan/rule_based_planner.cpp [hunk 1]
```diff
@@ -53,16 +53,19 @@ class ReturnBodyContext : public HierarchicalTreeVisitor {
...
     }
+
+    int pattern_comprehension_index = 0;
     for (auto &named_expr : body_.named_expressions) {
       output_symbols_.emplace_back(symbol_table_.at(*named_expr));
       named_expr->Accept(*this);
       named_expressions_.emplace_back(named_expr);
 
       // Pattern comprehension can be filled during named expression traversion
       if (auto it = pc_ops.find(named_expr->name_); it != pc_ops.end()) {
```
Comment (Josipmrden) [inline src/query/plan/rule_based_planner.cpp:64]: do you mg_assert ever that pc_ops is empty at the end?

File: src/query/plan/rule_based_planner.hpp [hunk 2]
```diff
@@ -33,8 +33,11 @@ struct PatternComprehensionData {
   PatternComprehensionData(std::shared_ptr<LogicalOperator> lop, Symbol res_symbol)
       : op(std::move(lop)), result_symbol(res_symbol) {}
 
+  PatternComprehensionData(PatternComprehension *pc, Symbol res_symbol)
+      : pattern_comprehension(pc), result_symbol(res_symbol) {}
+
   PatternComprehension *pattern_comprehension = nullptr;
-  std::shared_ptr<LogicalOperator> op;
+  std::shared_ptr<LogicalOperator> op = nullptr;
```
Comment (antoniofilipovic) [inline src/query/plan/rule_based_planner.hpp:40]: You don't need to define it as `nullptr` here, it should be empty: https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr 
Check (1)

**Review by antoniofilipovic — APPROVED** [review]
One comment, everything else looks fine

File: src/query/plan/rule_based_planner.hpp (hunk 2 above)
Comment (Darych) [reply src/query/plan/rule_based_planner.hpp:40]: Thank you, Fico! Will fix this :)

**Comment by kgolubic** [comment]
@Darych can you please provide me with Release notes and Docs PR?

**Pull request merged**